.github/workflows/build_and_test.yml

```diff
@@ -3,13 +3,9 @@ name: build-and-test
 on:
   push:
     branches: ['canary']
-  pull_request_target:
+  pull_request:
     types: [opened, synchronize]
 
-concurrency:
-  group: ${{ github.ref }}-build-and-test
-  cancel-in-progress: true
```

Comment:
@ijjk How about something like this:

```yaml
concurrency:
  group: ${{ github.workflow }}-${{ github.ref }}
  cancel-in-progress: ${{ github.ref != 'refs/heads/canary' }}
```

I'm curious what problem you were seeing with the native cancel?